src/Databases/DataLake/DatabaseDataLake.cpp

```diff
@@ -432,7 +438,7 @@ StoragePtr DatabaseDataLake::tryGetTableImpl(const String & name, ContextPtr con
 
     return std::make_shared<StorageObjectStorage>(
         configuration,
-        configuration->createObjectStorage(context_copy, /* is_readonly */ false),
+        lightweight ? nullptr : configuration->createObjectStorage(context_copy, /* is_readonly */ false),
```

Comment:
@alesapin : I don't like passing nullptr, but maybe we can avoid creating ObjectStorage if its lightweight. Do you have any better other suggestions ?
I feel if its lightweight we dont need ObjectStorage and another reason is lightweight will be missing credentials so it fails to authenticate (atleast for Azure)